src/main/kotlin/uk/gov/justice/digital/hmpps/dpssmoketest/service/PoeSmokeTestService.kt

```diff
@@ -3,23 +3,35 @@ package uk.gov.justice.digital.hmpps.dpssmoketest.service
 import org.springframework.stereotype.Service
 import reactor.core.publisher.Flux
 import uk.gov.justice.digital.hmpps.dpssmoketest.resource.SmokeTestResource.TestStatus
+import uk.gov.justice.digital.hmpps.dpssmoketest.resource.SmokeTestResource.TestStatus.TestProgress.COMPLETE
+import uk.gov.justice.digital.hmpps.dpssmoketest.resource.SmokeTestResource.TestStatus.TestProgress.INCOMPLETE
+import uk.gov.justice.digital.hmpps.dpssmoketest.resource.SmokeTestResource.TestStatus.TestProgress.SUCCESS
 
 @Service
 class PoeSmokeTestService(
-  private val prisonService: PrisonService
+  private val prisonService: PrisonService,
+  private val queueService: QueueService
+
 ) {
   fun runSmokeTest(testProfile: PoeTestParameters): Flux<TestStatus> {
+
+    // TODO (Should we ensure prisoner is in prison before attempting to release them?)
+    queueService.purgeQueue()
+
     return Flux.concat(
-      Flux.just(TestStatus("Will release prisoner ${testProfile.nomsNumber}", TestStatus.TestProgress.INCOMPLETE)),
-      // Trigger test by releasing prisoner
+      Flux.just(TestStatus("Will release prisoner ${testProfile.nomsNumber}", INCOMPLETE)),
       Flux.from(prisonService.triggerPoeReleaseTest(testProfile.nomsNumber)),
-      // Check for hmpps domain release event
-      Flux.from(prisonService.waitForEventToBeProduced(testProfile.nomsNumber)),
-
-      // Trigger test by recalling prisoner
-      // Flux.from(prisonService.triggerPoeRecallTest(it.nomsNumber)),
-      // Check for hmpps domain recall event
-      // prisonService.waitForEventToBeProduced(testProfile.nomsNumber),
+      Flux.from(
+        queueService.waitForEventToBeProduced(
+          "prison-offender-events.prisoner.released", testProfile.nomsNumber,
+          COMPLETE
+        )
+      ),
+      Flux.just(TestStatus("Will recall prisoner ${testProfile.nomsNumber}", INCOMPLETE)),
+      Flux.from(prisonService.triggerPoeRecallTest(testProfile.nomsNumber)),
+      Flux.from(queueService.waitForEventToBeProduced("prison-offender-events.prisoner.received", testProfile.nomsNumber, SUCCESS)),
     ).takeUntil(TestStatus::hasResult)
+
+    // TODO Question->shall we delete the queue?
```

Comment:
Hmmm it kind of depends upon how you receive messages from the queue - if you use a JMS listener then it will just constantly read the messages off the queue, but if you read the messages one by one with the SQS client then you might get tons of messages backed up in between smoke tests so purging the queue at the end makes no difference anyway.

Reply:
@mikehalmamoj It's not very explicit but I meant - should we delete the actual queue - if that's possible.